dali/test/python/test_dali_variable_batch_size.py

```diff
@@ -0,0 +1,622 @@
+    :param n_iter: Number of iterations in the epoch
+    :param sample_shape: If sample_shape is callable, shape of every sample will be determined by
+                         calling sample_shape. In this case, every call to sample_shape has to
+                         return a tuple of integers. If sample_shape is a tuple, this will be a
+                         shape of every sample.
+    :param lo:
+    :param hi:
+    :param dtype: 'int' for uint8 or 'float' for float32
+    :return: An epoch of data
+    """
+    batch_sizes = np.array([max_batch_size // 2, max_batch_size // 4, max_batch_size])
+
+    if isinstance(sample_shape, tuple):
+        size = sample_shape
+    elif inspect.isgeneratorfunction(sample_shape):
+        size = sample_shape()
+    else:
+        raise RuntimeError(
+            "`sample_shape` shall be either a tuple or a callable. Provide `(val,)` tuple for 1D shape")
+
+    if np.issubdtype(dtype, np.integer):
+        return [np.random.randint(lo, hi, size=(1, bs) + size, dtype=dtype) for bs in
+                batch_sizes]
+    elif np.issubdtype(dtype, np.float):
+        ret = (np.random.random_sample(size=(1, bs) + size) for bs in batch_sizes)
+        ret = map(lambda batch: (hi - lo) * batch + lo, ret)
+        ret = map(lambda batch: batch.astype(dtype), ret)
+        return list(ret)
+    else:
+        raise RuntimeError("Invalid type argument")
+
+
+def single_op_pipeline(max_batch_size, input_data, device, /, *, input_layout=None,
+                       operator_fn=None, **opfn_args):
+    pipe = Pipeline(batch_size=max_batch_size, num_threads=1, device_id=0)
+    with pipe:
+        input = fn.external_source(source=input_data, num_outputs=1, cycle=False, device=device,
+                                   layout=input_layout)
+        output = input[0] if operator_fn is None else operator_fn(input, device=device, **opfn_args)
+        pipe.set_outputs(output)
+    return pipe
+
+
+def run_pipeline(input_epoch, pipeline_fn, *, devices: list = ['cpu', 'gpu'], **pipeline_fn_args):
+    """
+    Verifies, if given pipeline supports iter-to-iter variable batch size
+
+    This function verifies only if given pipeline runs without crashing.
+    There is no qualitative verification. Use this for checking pipelines
+    based on random operators (as they can't be verifies against one another).
+
+    :param input_epoch: List of numpy arrays, where every item is a single batch
+    :param pipeline_fn: Function, that returns created (but not built) pipeline.
+                        Its signature should be (at least):
+                        pipeline_fn(max_batch_size, input_data, device, /, ...)
+    :param devices: Devices to run the check on
+    :param pipeline_fn_args: Additional args to pipeline_fn
+    """
+    for device in devices:
+        n_iter = len(input_epoch)
+        max_bs = max(batch.shape[1] for batch in input_epoch)
```

Comment:
```suggestion
        max_bs = max(batch.shape[0] for batch in input_epoch)
```
If you followed my advice to remove the leading dimension

Reply:
Done